packages/openapi-generator/src/parser/schema.ts

```diff
@@ -47,6 +47,14 @@ export function parseSchema(
     schema.properties ||
     'additionalProperties' in schema
   ) {
+    if (isAllOfSchema(schema)) {
+      return {
+        ...(schema.allOf?.length &&
+          parseXOfSchema(schema, refs, 'allOf', options)),
+        ...((schema.properties || 'additionalProperties' in schema) &&
+          parseObjectSchema(schema, refs, options))
+      };
+    }
```

Comment:
[req] I still believe combining `allOf` and `type: object` is just incorrect, therefore if both are set and we know a few cases where services do that, I would allow that and treat it as `allOf` only. Parsing other properties and handling them a certain way is just guess work and as far as I know we haven't seen examples of that before, did we?

Therefore my suggestion is to just move this down below `anyOf` so objects are prioritized lower.

Reply:
[pp] I think it could be helpful to have a warning when generating this (incorrect) type. What do you think?

Reply:
Added info log if allOf and properties are at the same level. It is restructured into the "inheritance" type as discussed